Unclear how to use build_nginx

The [README](https://github.com/heroku/heroku-buildpack-nginx#customizable-nginx-compile-options) mentions that it is possible to customize nginx's compile options via the [https://github.com/heroku/heroku-buildpack-nginx/blob/master/scripts/build_nginx](build_nginx) script.

The [script says](https://github.com/heroku/heroku-buildpack-nginx/blob/master/scripts/build_nginx#L2-L10) it's designed to be run in a Heroku web dyno and the final binary downloaded for vendoring. However, it is light on details as to how to go about this.

Looking through the script's history, I've found a call to setup a Python web server but this was removed in https://github.com/heroku/heroku-buildpack-nginx/commit/a51ae6300ca6b80a82f3ec0f446025229edfd8da#diff-c1594fe8a33425936bb1b25a1f142c6b. Perhaps a red herring but the loop to keep the bash script running was also removed in https://github.com/heroku/heroku-buildpack-nginx/commit/e64133f2f3ebccdcb8bbdca422294dfa6ecd6897#diff-c1594fe8a33425936bb1b25a1f142c6b.

What I've attempted so far is:
* clone the repository
* make the change I wanted (including `--with-http_geoip_module` in the `./configure` call)
* added back in the python server (creating a `requirements.txt` file so Heroku would use the Python buildpack)
* create a `Profile` with `web: scripts/build_nginx`
* and push to a new Heroku app

I'm able to deploy by my [logs.txt](https://github.com/heroku/heroku-buildpack-nginx/files/5113092/logs.txt) (relevant portion duplicated below) say the app crashes.

Googling has also hit a dead end for the moment so I'm hopeful that whatever I'm missing is more obvious to those with more experience than me.

```
2020-08-22T21:40:45.033008+00:00 app[web.1]: checking for OS
2020-08-22T21:40:45.037078+00:00 app[web.1]: + Linux 4.4.0-1076-aws x86_64
2020-08-22T21:40:46.670273+00:00 app[web.1]: checking for C compiler ... not found
2020-08-22T21:40:46.679991+00:00 app[web.1]: 
2020-08-22T21:40:46.679993+00:00 app[web.1]: ./configure: error: C compiler cc is not found
2020-08-22T21:40:46.679994+00:00 app[web.1]: 
2020-08-22T21:40:46.689548+00:00 app[web.1]: make: *** No rule to make target 'install'.  Stop.
2020-08-22T21:40:46.747961+00:00 app[web.1]: checking for OS
2020-08-22T21:40:46.751417+00:00 app[web.1]: + Linux 4.4.0-1076-aws x86_64
2020-08-22T21:40:46.772664+00:00 app[web.1]: checking for C compiler ... not found
2020-08-22T21:40:46.776255+00:00 app[web.1]: 
2020-08-22T21:40:46.776302+00:00 app[web.1]: ./configure: error: C compiler cc is not found
2020-08-22T21:40:46.776325+00:00 app[web.1]: 
2020-08-22T21:40:46.779272+00:00 app[web.1]: make: *** No rule to make target 'install'.  Stop.
2020-08-22T21:40:46.784637+00:00 app[web.1]: cp: cannot stat '/tmp/nginx/sbin/nginx': No such file or directory
2020-08-22T21:40:46.786734+00:00 app[web.1]: cp: cannot stat '/tmp/nginx-debug/sbin/nginx': No such file or directory
2020-08-22T21:40:46.788976+00:00 app[web.1]: cp: cannot stat '/tmp/nginx/conf/mime.types': No such file or directory
2020-08-22T21:40:46.793573+00:00 app[web.1]: ./
2020-08-22T21:40:46.797268+00:00 app[web.1]: cp: missing destination file operand after '/tmp/nginx-.tgz'
2020-08-22T21:40:46.797275+00:00 app[web.1]: Try 'cp --help' for more information.
2020-08-22T21:40:46.799339+00:00 app[web.1]: cp: missing destination file operand after '/tmp/nginx/sbin/nginx'
2020-08-22T21:40:46.799344+00:00 app[web.1]: Try 'cp --help' for more information.
2020-08-22T21:40:46.836898+00:00 heroku[web.1]: Process exited with status 0
2020-08-22T21:40:46.876657+00:00 heroku[web.1]: State changed from starting to crashed
2020-08-22T21:40:48.515167+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=custom-nginx.herokuapp.com request_id=2ef22723-daf3-4645-96ab-c7cdc4ea47d2 fwd="97.77.198.5" dyno= connect= service= status=503 bytes= protocol=https
2020-08-22T21:40:49.319575+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=custom-nginx.herokuapp.com request_id=9117a2eb-2bc6-4abe-bf9f-792e561313a9 fwd="97.77.198.5" dyno= connect= service= status=503 bytes= protocol=https
```

--- a/changelog.md
+++ b/changelog.md
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.5.1] - 2020-08-22
+### Changes
+- [readme.md] Updated custom build instructions
+- [scripts/build_nginx] Updated outdated comments
+
 ## [1.5] - 2020-07-20
 ### Added
 - [all] move to tgz to distribute binaries

--- a/readme.md
+++ b/readme.md
@@ -110,15 +110,25 @@ You can provide your own NGINX config by creating a file named `nginx.conf.erb` 
 
 ### Customizable NGINX Compile Options
 
-See [scripts/build_nginx](scripts/build_nginx) for the build steps. Configuring is as easy as changing the "./configure" options.
+This requires a clone of this repository and [Docker](https://www.docker.com/). All you need to do is have Docker setup and running on your machine. The [`Makefile`](Makefile) will take care of the rest.
 
-You can run the builds in a [Docker](https://www.docker.com/) container:
+Configuring is as easy as changing the options passed to `./configure` in [scripts/build_nginx](scripts/build_nginx).
+
+Run the builds in a container via:
 
 ```
-$ make build # It outputs the latest builds to bin/cedar-*
+$ make build
 ```
 
-To test the builds:
+The binaries will be packed into `tar` files and placed in the repository's root directory. Commit the changes and push your repository.
+
+Finally update your app to use your custom buildpack on Heroku either at https://dashboard.heroku.com/apps/#{YOUR_APP_NAME}/settings or via the Heroku CLI via:
+
+```
+heroku buildpacks:set #{YOUR_GIT_REPO_CLONE}
+```
+
+To test the builds locally:
 
 ```
 $ make shell

--- a/scripts/build_nginx
+++ b/scripts/build_nginx
@@ -1,13 +1,8 @@
 #!/bin/bash
-# Build NGINX and modules on Heroku.
-# This program is designed to run in a web dyno provided by Heroku.
-# We would like to build an NGINX binary for the builpack on the
-# exact machine in which the binary will run.
-# Our motivation for running in a web dyno is that we need a way to
-# download the binary once it is built so we can vendor it in the buildpack.
-#
-# Once the dyno has is 'up' you can open your browser and navigate
-# this dyno's directory structure to download the nginx binary.
+# Build NGINX and modules for Heroku.
+# This script is designed to run in a Heroku Stack Docker
+# image. More information on the Heroku Stack can be found
+# at https://devcenter.heroku.com/articles/stack
 
 NGINX_VERSION=${NGINX_VERSION-1.18.0}
 PCRE_VERSION=${PCRE_VERSION-8.44}
@@ -41,6 +36,7 @@ echo "Downloading $zlib_url"
 echo "Downloading $uuid4_url"
 (cd nginx-${NGINX_VERSION} && curl -L $uuid4_url | tar xvz )
 
+# This will build `nginx`
 (
   cd nginx-${NGINX_VERSION}
   ./configure \
@@ -55,6 +51,7 @@ echo "Downloading $uuid4_url"
   make install
 )
 
+# This will build `nginx-debug`
 (
   cd nginx-${NGINX_VERSION}
   ./configure \

